test/spec/token.js

```diff
@@ -112,6 +112,63 @@ define(function(require) {
       });
     });
 
+    it('allows multiple iframes simultaneously', function (done) {
+      return oauthUtil.setupSimultaneousPostMessage()
+      .then(function(context) {
+        // mock frame creation
+        var body = document.getElementsByTagName('body')[0];
+        var origAppend = body.appendChild;
+        spyOn(body, 'appendChild').and.callFake(function (el) {
+          if (el.tagName === 'IFRAME') {
+            // Remove the src so it doesn't actually load
+            el.src = '';
+            return origAppend.call(this, el);
+          }
+          return origAppend.apply(this, arguments);
+        });
+
+        // ensure that no iframes are open
+        var iframes = document.getElementsByTagName('IFRAME');
+        expect(iframes.length).toBe(0);
+
+        // getWithoutPrompt, but don't resolve
+        var firstPrompt = context.client.token.getWithoutPrompt({
+          sessionToken: 'testSessionToken',
+          state: oauthUtil.mockedState,
+          nonce: oauthUtil.mockedNonce
+        });
+
+        // getWithoutPrompt, but don't resolve
+        var secondPrompt = context.client.token.getWithoutPrompt({
+          sessionToken: 'testSessionToken2',
+          state: oauthUtil.mockedState2,
+          nonce: oauthUtil.mockedNonce2
+        });
+
+        // assert that two iframes are open
+        expect(iframes.length).toBe(2);
```

Comment:
Interesting - this works? I.e. I'm surprised that we can do a document.getElementsByTagName, which returns an array of elements, and that array of elements will dynamically update without having to search for the iframes again.